### DISCUSSION TITLE
[next/image] Disable auto webp.

### DISCUSSION THREAD

**Discussion by myWsq** [vercel/next.js#25136] [Help] [answered]
**Describe the feature you'd like to request**

The format of the image is not in the url parameter, which makes it tricky to access the CDN. Could you provide an option to disable auto webp?

**Describe the solution you'd like**

```js
module.exports = {
  images: {
    webp: false
  },
}
```

**Describe alternatives you've considered**

_No response_

**Comment by timneutkens** [comment]
There is no clear benefit in providing such an option.

> makes it tricky to access the CDN

CDNs support serving content in webp so this shouldn't be a problem.

**Comment by thesiti92** [comment]
@timneutkens thats not what hes saying. hes saying if you use a cdn then it will cache webp images and serve it to older browsers like safari because theres no image format in the url. we cant use CloudFlare with next/image as a result

**Comment by avisra** [comment]
This ticket needs to be opened back up. it is 100% valid @timneutkens.

NextJS delivers webp format automatically. For those of us using a CDN (like Cloudflare), it will cache those assets from the origin. Which then causes non-webp-supporting browsers (Safari on Mac) to not render images.

We should be able to instruct next/image to not convert to webp so we can allow our CDN to do it.

**Comment by myWsq** [comment]
Many CDNs don't support automatic determination of image type, so I think should add a format parameter to the link to use with `<source type="image/webp">`

**Reply by ascorbic** [reply]
This is a great idea. I've had a think about it and have a more detailed proposal: #25393

**Comment by avisra** [comment]
This seems like a fairly simple change to make and should be low impact - especially if the default is to leave webp enabled.

**Comment by st4ng** [comment]
@thesiti92 @avisra If you want to cache next/image (webp + fallback) with a CDN and need a workaround right now, you can just rewrite your url with some additional query parameter for requests from browsers that don't support webp. For example with Cloudflare just use the following transform rule:

> **Expression Editor**:
> `(http.request.uri.path eq "/_next/image" and not any(http.request.headers["accept"][*] contains "image/webp"))`
> **Rewrite Query to:**
> Dynamic | `concat(http.request.uri.query, "&webp=false")`

I wrote a short [blog post](https://st4ng.medium.com/how-to-use-next-js-image-optimization-with-cloudflare-569da7b3ddc6) if you need more info

**Reply by reigio** [reply]
Hello @st4ng 
Came across your blog post but seems it is not valid anymore. When trying to set it up CDN (CloudFlare) presents an error "not a valid value for query because expression is invalid".
Seems CF has changed something, for example http.request.headers is not in the list of parameters.

**Reply by st4ng** [reply]
Hey @reigio,

thanks for letting me know! I gave it a quick check and seems like the issue is not with Coudflare, but the quotation marks are messed up in the blog post and also in my answer above :sweat_smile: They are `“` instead of `"`, that's why Cloudflare is complaining if you copy paste it.

I already fixed it in my answer above, so you can copy paste from there. I think apart from that, it should still work. Haven't been using it myself for a while though.

**Comment by styfle** [comment]
If you have a CDN in front of Next.js, upgrading to the latest version should fix the caching issue since the Image Optimization API now provides the `Vary` header since #26788.

This indicates to any downstream cache that the response body varies based on the request's `Accept` header.

**Reply by remorses** [reply]
Cloudflare only uses Vary on Pro plans

**Comment by styfle** [comment, marked as answer]
You can disable it by configuring `formats` in next.config.js 

```js
module.exports = {
  images: {
    formats: [],
  },
};
```

The empty array makes sure nothing will match the `Accept` header, therefore the source image format will be preserved for the optimized image.

https://nextjs.org/docs/pages/api-reference/components/image#formats

This should be a last resort however and you should instead fix your CDN to observe the `Vary` header and pass through the `Accept` header like the [previous answer](https://github.com/vercel/next.js/discussions/25136#discussioncomment-1329501).
